lib/cloudant.js

```diff
@@ -742,23 +736,41 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   debug('Cloudant.prototype.updateOrCreate %j %j', model, data);
   var self = this;
   var idName = self.idName(model);
-  var id = data[idName];
+  var mo = self.selectModel(model);
+  var id = data[idName].toString();
+
+  // Callback handler for both create calls.
+  var createHandler = function(err, id) {
+    if (err) return cb(err);
+    mo.db.get(id, function(err, doc) {
+      if (err) return cb(err);
+      cb(err, self.fromDB(model, mo, doc), {isNewInstance: true});
+    });
+  };
+
   if (id) {
-    self.updateAttributes(model, id, data, {}, function(err, docs) {
-      if (err && err.statusCode && err.statusCode === 404) {
-        self.create(model, data, {}, function(err) {
+    self.exists(model, id, {}, function(err, count) {
+      if (err) return cb(err);
+      else if (count > 0) {
+        self.getCurrentRevision(model, id, function(err, rev) {
```

Comment:
This is completely wrong, and **not** what I had in my PR originally! This will overrwrite the provided `_.rev` from the user! :-1: